src/plots/plot.ts

```diff
@@ -155,28 +155,30 @@ export class Plot extends Component {
   }
 
   protected _bindProperty(property: string, value: any, scale: Scale<any, any>) {
-    this._bind(property, value, scale, this._propertyBindings, this._propertyExtents);
-    this._updateExtentsForProperty(property);
+    let updateExtentsForKey = () => this._updateExtentsForProperty(property);
+    this._bind(property, value, scale, this._propertyBindings, this._propertyExtents, updateExtentsForKey);
   }
 
   private _bindAttr(attr: string, value: any, scale: Scale<any, any>) {
-    this._bind(attr, value, scale, this._attrBindings, this._attrExtents);
-    this._updateExtentsForAttr(attr);
+    let updateExtentsForKey = () => this._updateExtentsForAttr(attr);
+    this._bind(attr, value, scale, this._attrBindings, this._attrExtents, updateExtentsForKey);
   }
 
   private _bind(key: string, value: any, scale: Scale<any, any>,
-                    bindings: d3.Map<Plots.AccessorScaleBinding<any, any>>, extents: d3.Map<any[]>) {
+                bindings: d3.Map<Plots.AccessorScaleBinding<any, any>>, extents: d3.Map<any[]>,
+                updateExtentsForKey: (key: string) => void) {
     let binding = bindings.get(key);
     let oldScale = binding != null ? binding.scale : null;
 
+    bindings.set(key, { accessor: d3.functor(value), scale: scale });
+    updateExtentsForKey(key);
```

Comment:
yeah, I tried that first, but it actually produces a lot more code and looks messier
because you need to retrieve and store `oldScale` at `_bindProperty` and `_bindAttr` level
I could do it that way to avoid API signature change if it matters for protected functions